### PR TITLE
Fix incorrect conditions in eng/common/templates/job/job.yml

### PR DESCRIPTION
**Pull request by ulisesh**
Arcade issue https://github.com/dotnet/arcade/issues/12393

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Review by garath — DISMISSED** [review]

File: eng/common/templates/job/job.yml [hunk 1]
```diff
@@ -134,7 +134,7 @@ jobs:
   - ${{ if and(eq(parameters.runAsPublic, 'false'), eq(variables['System.TeamProject'], 'internal')) }}:
     - task: NuGetAuthenticate@0
 
-  - ${{ if or(eq(parameters.artifacts.download, 'true'), ne(parameters.artifacts.download, '')) }}:
+  - ${{ if or(eq(parameters.artifacts.download, 'true'), eq(parameters.artifacts.download, '')) }}:
```
Comment (MattGal) [inline eng/common/templates/job/job.yml:137]: This makes "false" work, but still does the behavior in the case of a missing parameter.  I strongly feel it should be just:
```
 ${{ if eq(parameters.artifacts.download, 'true') }}
```

Comment (ulisesh) [reply eng/common/templates/job/job.yml]: Changed, I'll update the issue

**Review by MattGal — APPROVED** [review]

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Comment by RussKie** [comment]
On a second thought (after speaking with @MattGal) I think it was wrong to omit the second part of the condition. That is, it's probably not uncommon for consumers to _not_ specify the "root" node (such as `parameters.artifacts.publish.logs`) and only specify a subnode (e.g., `parameters.artifacts.publish.logs.name`). This means, that with this fix the existing definitions, such as this, will stop working:
https://github.com/dotnet/arcade/blob/034aa38aefca48451af44494ad5c284780db90da/eng/validate-sdk.yml#L9-L19

This is why we need the second condition that checks for the empty string.

**Pull request merged**